### PR TITLE
zohoInvoice connector added

### PR DESCRIPTION
**Pull request by ashalya**

### CODE REVIEW

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/zoho.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.carbon.connector;
+
+import org.apache.synapse.MessageContext;
+import org.wso2.carbon.connector.core.*;
+
+public class zoho extends AbstractConnector {
```
Comment (keerthu) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/zoho.java]: You didn't use this class anywhere. So remove this.

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/config/init.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+-->
+    <parameter name="apiUrl" description="URL of the ZohoInvoice API"/>
+    <sequence>
+        <property name="uri.var.apiUrl" expression="$func:apiUrl"/>
+        <property name="uri.var.apiUrl" value="https://invoice.zoho.com/api/v3"/>
```
Comment (keerthu) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/config/init.xml]: Remove the hardcoded value.

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/contacts/createAContact.xml [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+               description="Default estimate email template id used for this contact while sending estimates"/>
+    <parameter name="creditnoteEmailTemplateId"
+               description="Default credit note email template id used for this contact while sending credit notes"/>
+    <parameter name="invoiceImailTemplateId"
+               description="Default invoice email template id used for this contact while sending invoices"/>
+    <sequence>
+        <property name="uri.var.authToken" expression="$func:authToken"/>
+        <property name="uri.var.contactName" expression="$func:contactName"/>
+        <property name="uri.var.companyName" expression="$func:companyName"/>
+        <property name="uri.var.companyName" expression="$func:companyName"/>
```
Comment (keerthu) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/contacts/createAContact.xml]: Remove duplicate property.

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/creditNotes/createACreditNote.xml [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        <property name="uri.var.contactPersons" expression="$func:contactPersons"/>
+        <property name="uri.var.creditNoteNumber" expression="$func:creditNoteNumber"/>
+        <property name="uri.var.referenceNumber" expression="$func:referenceNumber"/>
+        <property name="uri.var.templateId" expression="$func:templateId"/>
+        <property name="uri.var.date" expression="$func:date"/>
+        <property name="uri.var.exchangeRate" expression="$func:exchangeRate"/>
+        <property name="uri.var.notes" expression="$func:notes"/>
+        <property name="uri.var.terms" expression="$func:terms"/>
+        <property name="zohoParameters" value=""/>
+        <property name="zohoParameters1" value=""/>
```
Comment (keerthu) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/creditNotes/createACreditNote.xml]: Use meaningful name.

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/estimates/UpdateBillingAddress.xml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+-->
+<template name="UpdateBillingAddress" xmlns="http://ws.apache.org/ns/synapse">
+    <parameter name="authtoken" description="API key"/>
+    <parameter name="estimate_id" description="Id of the estimate"/>
```
Comment (keerthu) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/estimates/UpdateBillingAddress.xml]: Use camel case.

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/estimates/UpdateShippingAddress.xml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+-->
+<template name="UpdateShippingAddress" xmlns="http://ws.apache.org/ns/synapse">
+    <parameter name="authtoken" description="API key"/>
+    <parameter name="estimate_id" description="Id of the estimate"/>
```
Comment (keerthu) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/estimates/UpdateShippingAddress.xml]: Use camel case.

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/estimates/updateAnExistingEstimate.xml [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+    specific language governing permissions and limitations
+    under the License.
+-->
+<template name="updateAnExistingEstimate" xmlns="http://ws.apache.org/ns/synapse">
+    <parameter name="authToken" description="API key"/>
+    <parameter name="customerId" description="ID of the customer the estimate has to be created"/>
+    <parameter name="contactPersons" description="Array of contact person(s) for whom estimate has to be sent"/>
+    <parameter name="lineItems" description="Line items of an estimate"/>
+    <parameter name="templateId" description="ID of the pdf template associated with the estimate"/>
+    <parameter name="estimate_number" description="Mandatory if auto number generation is disabled"/>
```
Comment (keerthu) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/estimates/updateAnExistingEstimate.xml]: Use camel case.

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/invoices/cancelTheWriteOffAnInvoice.xml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+-->
+<template name="cancelTheWriteOffAnInvoice" xmlns="http://ws.apache.org/ns/synapse">
+    <parameter name="authToken" description="API key"/>
+    <parameter name="invoiceId"/>
```
Comment (keerthu) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/invoices/cancelTheWriteOffAnInvoice.xml]: Missed the description.

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/items/getADetailOfAnItem.xml [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+-->
+<template name="getADetailOfAnItem" xmlns="http://ws.apache.org/ns/synapse">
+    <parameter name="authToken" description="API key"/>
+    <parameter name="itemsId"/>
```
Comment (keerthu) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/items/getADetailOfAnItem.xml]: Missed the description.

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/zoho.java (hunk 1 above)
Comment (ashalya) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/java/org/wso2/carbon/connector/zoho.java]: fixed in https://github.com/wso2/esb-connectors/pull/450

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/config/init.xml (hunk 2 above)
Comment (ashalya) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/config/init.xml]: fixed in https://github.com/wso2/esb-connectors/pull/450

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/contacts/createAContact.xml (hunk 3 above)
Comment (ashalya) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/contacts/createAContact.xml]: fixed in https://github.com/wso2/esb-connectors/pull/450

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/creditNotes/createACreditNote.xml (hunk 4 above)
Comment (ashalya) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/creditNotes/createACreditNote.xml]: fixed in https://github.com/wso2/esb-connectors/pull/450

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/estimates/UpdateBillingAddress.xml (hunk 5 above)
Comment (ashalya) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/estimates/UpdateBillingAddress.xml]: fixed in https://github.com/wso2/esb-connectors/pull/450

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/estimates/UpdateShippingAddress.xml (hunk 6 above)
Comment (ashalya) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/estimates/UpdateShippingAddress.xml]: fixed in https://github.com/wso2/esb-connectors/pull/450

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/estimates/updateAnExistingEstimate.xml (hunk 7 above)
Comment (ashalya) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/estimates/updateAnExistingEstimate.xml]: fixed in https://github.com/wso2/esb-connectors/pull/450

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/invoices/cancelTheWriteOffAnInvoice.xml (hunk 8 above)
Comment (ashalya) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/invoices/cancelTheWriteOffAnInvoice.xml]: fixed in https://github.com/wso2/esb-connectors/pull/450

File: zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/items/getADetailOfAnItem.xml (hunk 9 above)
Comment (ashalya) [inline zohoInvoice/zohoInvoice-connector/zohoInvoice-connector-1.0.0/org.wso2.carbon.connector/src/main/resources/items/getADetailOfAnItem.xml]: fixed in https://github.com/wso2/esb-connectors/pull/450

**Pull request closed without merging**